master: kickstart: support firewall --use-system-defaults

dustymabe opened the pull request:
Needed for [1] where we would like to include firewalld
and configure firewalld in Atomic Host (in the ostree) and
have Anaconda leave the delivered "defaults" in place. The
action here is to do nothing if the user specified
--use-system-defaults.

[1] https://pagure.io/atomic-wg/issue/401

dustymabe commented:
related to: https://github.com/rhinstaller/pykickstart/pull/203

I did not test this one thoroughly like I did the patches to to the f27 branches because the transition to pykickstart 3 hasn't happened yet. I will make sure to be involved when that happens in rawhide.

cgwalters commented:
Seems sane to me.

rvykydal commented:
We should also increment required pykickstart version in anaconda.spec.in

jkonecny12 commented:
Jenkins, it's ok to test.

jkonecny12 commented:
Tests are failing because of missing `F28_Firewall` class.

dustymabe commented:
> Tests are failing because of missing F28_Firewall class.

so we need to wait for the new pykickstart first?

jkonecny12 commented:
Yes we should wait for pykickstart first, otherwise we would be blocked to make a new release and all the tests will be failing.

dustymabe commented:
the [pykickstart PR](https://github.com/rhinstaller/pykickstart/pull/203) for master was merged

jkonecny12 commented:
jenkins, test this please

dustymabe commented:
> Could you increase pykickstart version in the spec file please.

I'm guessing the version will be `%define pykickstartver 2.44-1`. rebased and added that.

dustymabe commented:
> And now we need to wait till new pykickstart version will be released. After that this PR can be merged and new Anaconda released.

@clumens ^^ FYI

@jkonecny12 - note that the commits from the other branches have **(cherry picked from commit 8d06406)** messages in them. I'm not sure if merging will preserve the commit hash or not, so let's be careful when we do merge this set of PRs. I'll rebase the other two branches if needed.

dustymabe commented:
https://github.com/rhinstaller/pykickstart/releases/tag/r2.44-1 is out

jkonecny12 commented:
jenkins, test this please

jkonecny12 commented:
> @jkonecny12 - note that the commits from the other branches have (cherry picked from commit 8d06406) messages in them. I'm not sure if merging will preserve the commit hash or not, so let's be careful when we do merge this set of PRs. I'll rebase the other two branches if needed.

I think it should be fine for Fedora. The only potential problem I can think of is bugzilla connection to bugs but that is done by Bodhi which should look on the correct branch and isn't enabled on Rawhide.
